kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableSchemaEvolutionSuite.scala

```diff
@@ -1196,31 +1200,63 @@ class DeltaTableSchemaEvolutionSuite extends DeltaTableWriteSuiteBase with Colum
-      val table = Table.forPath(engine, tablePath)
-      val initialSchema = new StructType()
-        .add("data", StringType.STRING, true)
-        .add("clustering-col", IntegerType.INTEGER, true)
-
-      createEmptyTable(
-        engine,
-        tablePath,
-        initialSchema,
-        clusteringCols = List(new Column("clustering-col")),
-        tableProperties = Map(
-          TableConfig.COLUMN_MAPPING_MODE.getKey -> "id",
-          TableConfig.ICEBERG_COMPAT_V2_ENABLED.getKey -> "true"))
-
-      val currentSchema = table.getLatestSnapshot(engine).getSchema
-      val newSchema = new StructType()
-        .add("data", StringType.STRING, true, currentSchema.get("data").getMetadata)
-
-      assertSchemaEvolutionFails[KernelException](
-        table,
-        engine,
-        newSchema,
-        "Cannot drop clustering column clustering-col")
+  val primitiveSchemaWithClusteringColumn = new StructType()
+    .add("clustering_col", IntegerType.INTEGER)
+    .add("data", IntegerType.INTEGER)
+
+  val nestedSchemaWithClusteringColumn = new StructType()
+    .add(
+      "struct",
+      new StructType()
+        .add("clustering_col", IntegerType.INTEGER)
+        .add("data", IntegerType.INTEGER),
+      true)
+
+  private val updatedSchemaWithDroppedClusteringColumn = Tables.Table(
+    ("schemaBefore", "updatedSchemaWithDroppedClusteringColumn", "clusteringColumn"),
+    (
+      primitiveSchemaWithClusteringColumn,
+      new StructType()
+        .add(
+          "data",
+          StringType.STRING,
+          true,
+          primitiveSchemaWithClusteringColumn.get("data").getMetadata),
+      new Column("clustering_col")),
+    (
+      nestedSchemaWithClusteringColumn,
+      new StructType()
+        .add(
+          "struct",
+          new StructType()
+            .add(
+              "data",
+              IntegerType.INTEGER,
+              nestedSchemaWithClusteringColumn.get("struct").getDataType
+                .asInstanceOf[StructType].get("data").getMetadata),
+          true,
+          nestedSchemaWithClusteringColumn.get("struct").getMetadata),
+      new Column(Array("struct", "clustering_col"))))
+
+  test("Cannot drop clustering column") {
+    forAll(updatedSchemaWithDroppedClusteringColumn) {
+      (schemaBefore, schemaAfter, clusteringColumn) =>
+        withTempDirAndEngine { (tablePath, engine) =>
+          val table = Table.forPath(engine, tablePath)
+          createEmptyTable(
+            engine,
+            tablePath,
+            schemaBefore,
+            clusteringCols = List(clusteringColumn),
+            tableProperties = Map(
+              TableConfig.COLUMN_MAPPING_MODE.getKey -> "id",
+              TableConfig.ICEBERG_COMPAT_V2_ENABLED.getKey -> "true"))
+          assertSchemaEvolutionFails[KernelException](
+            table,
+            engine,
+            schemaAfter,
+            "Cannot drop clustering column clustering_col")
```

Comment:
I added a ToDo to make sure our failure messages have the full path name, it's something we should really fix now :(. For now I've left it as is.